File: client/clojure/src/cdisc_library_api/specs/cdash_domain_field.clj
(ns cdisc-library-api.specs.cdash-domain-field
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.cdash-domain-field-links :refer :all]
            )
  (:import (java.io File)))


(def cdash-domain-field-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :definition) string?
   (ds/opt :domainSpecific) string?
   (ds/opt :questionText) string?
   (ds/opt :prompt) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :implementationNotes) string?
   (ds/opt :mappingInstructions) string?
   (ds/opt :_links) cdash-domain-field-links-spec
   })

(def cdash-domain-field-spec
  (ds/spec
    {:name ::cdash-domain-field
     :spec cdash-domain-field-data}))

File: src/main/scala/org/openapitools/client/api/CdashDomainField.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CdashDomainField._

case class CdashDomainField (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
definition: Option[String],
domainSpecific: Option[String],
questionText: Option[String],
prompt: Option[String],
simpleDatatype: Option[String],
implementationNotes: Option[String],
mappingInstructions: Option[String],
links: Option[CdashDomainFieldLinks])

object CdashDomainField {
  import DateTimeCodecs._

  implicit val CdashDomainFieldCodecJson: CodecJson[CdashDomainField] = CodecJson.derive[CdashDomainField]
  implicit val CdashDomainFieldDecoder: EntityDecoder[CdashDomainField] = jsonOf[CdashDomainField]
  implicit val CdashDomainFieldEncoder: EntityEncoder[CdashDomainField] = jsonEncoderOf[CdashDomainField]
}
